jina/hubble/hubio.py

```diff
@@ -584,13 +584,54 @@ def _fetch_meta(
             md5sum=resp['package']['md5'],
         )
 
+    def _pull_with_progress(self, log_streams, console):
+        from rich.progress import Progress, DownloadColumn, BarColumn
+
+        with Progress(
+            "[progress.description]{task.description}",
+            BarColumn(),
+            DownloadColumn(),
+            "[progress.percentage]{task.percentage:>3.0f}%",
+            console=console,
+            transient=True,
+        ) as progress:
+            try:
+                tasks = {}
+                for log in log_streams:
+                    if 'status' not in log:
+                        continue
+                    status = log['status']
+                    status_id = log.get('id', None)
+                    pg_detail = log.get('progressDetail', None)
+
+                    if (pg_detail is None) or (status_id is None):
+                        console.print(status)
+                        continue
+
+                    if status_id not in tasks:
+                        tasks[status_id] = progress.add_task(status, total=0)
+
+                    task_id = tasks[status_id]
+
+                    if ('current' in pg_detail) and ('total' in pg_detail):
+                        progress.update(
+                            task_id,
+                            completed=pg_detail['current'],
+                            total=pg_detail['total'],
+                            description=status,
+                        )
+                    elif not pg_detail:
+                        progress.update(task_id, advance=0, description=status)
+
+            except Exception as ex:
+                raise ex
```

Comment:
if it just simply catches and raises then we do not need this `try` block

Reply:
@hanxiao `try` block has been removed